### PR TITLE
fixes #4380 [vSphere] allow selection of guest OS

### PR DESCRIPTION
**Pull request by mmatuska**
This allows to set the Guest OS type when creating a VMWare VM and saving it in compute profiles, too.
In the future we may consider doing a pre-selection based on Foreman operating system settings if creating a new VM, but we have to makte this work smoothly with compute profiles.

### CODE REVIEW

**Comment by angystardust** [comment]
I was waiting for that! Thank you @mmatuska! :+1: 
It would be great to include this patch in 1.4.1 release...

**Comment by mmoll** [comment]
WFM

**Comment by domcleal** [comment]
@mmatuska sorry for the delay.  You mentioned on IRC that there's a fixed enum in rbvmomi too (VirtualMachineGuestOsIdentifier).  Can we use this in addition to the descriptions present in Foreman, so even if we don't have a human-readable OS description, we at least present as many options as the underlying library supports?  I'm slightly concerned that we'll have a lag updating OS descriptions after rbvmomi adds them.

**Comment by mmatuska** [comment]
I have updated the PR to read the types enum from rbvmomi

File: app/models/compute_resources/foreman/model/vmware.rb [hunk 1]
```diff
@@ -83,6 +83,130 @@ def scsi_controller_types
...
+        "vmkernelGuest" => "VMWare ESX 4.x",
+        "vmkernel5Guest" => "VMWare ESXi 5.x",
+        "otherGuest" => "Other (32-bit)",
+        "otherGuest64" => "Other (64-bit)"
+      }
+    end
+
+    def guest_types
+      types = { }
+      RbVmomi::VIM::VirtualMachineGuestOsIdentifier.values.each do |value|
```
Comment (domcleal) [inline app/models/compute_resources/foreman/model/vmware.rb]: Nitpick, slightly shorter style would be:

<pre>
def guest_types
  RbVmomi::VIM::VirtualMachineGuestOsIdentifier.values.compact.map { |v| guest_types_descriptions[v] || v }
end
</pre>


(compact removes nils)

**Comment by domcleal** [comment]
Looks good, I'll give it a try.

**Comment by domcleal** [comment]
Merged as 507ebfd for Foreman 1.5.0, thanks @mmatuska!

**Pull request closed without merging**